ui/src/app/cohort-search/modifier-page/modifier-page.component.tsx

```diff
@@ -306,10 +307,19 @@ export const ModifierPage = withCurrentWorkspace()(
         formState.push(encounters);
         this.setState({visitCounts});
       }
+      formState = this.modifiersForSurvey;
       this.setState({formState});
       this.getExisting();
     }
 
+    get modifiersForSurvey() {
+      const {formState} = this.state;
+      const {searchContext: {domain}} = this.props;
+      if (domain === DomainType.SURVEY) {
+        return formState.filter(form => SURVEY_MODIFIERS.indexOf(form.name) > -1);
+      }
+      return formState;
+    }
     getExisting() {
       const {searchContext} = this.props;
       const {formState} = this.state;
```

Comment:
We also need to filter the form here for Surveys. When I added a Survey item, then went back in to edit the item, all three modifier fields showed in the form.